### PR TITLE
Create class to track websockets instances

### PR DESCRIPTION
**Pull request by ghost**
I reorganized the global variables into a class called `Connections`. It stores a dict of `WebsocketsCommonProtocol` keyed 
to usernames. Methods exist to add and delete users, and the class keeps track of the newest and most recently departed user. 
I revised the existing coroutines to work with the class, but they mostly have the same structure.

### CODE REVIEW

File: server/__main__.py [hunk 1]
```diff
@@ -2,74 +2,72 @@
 import json
 
 import websockets
+from connections import Connections
```
Comment (Kronifer) [inline server/__main__.py]:
```suggestion
from .connections import Connections
```
Making this a relative import makes more sense in this case

File: server/connections.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    @property
+    def newest_user(self) -> None | str:
+        """
+        Newest user property
+
+        Returns username of most recently joined user,
+        None if no one has joined yet
+        """
+        return self._newest_user
```
Comment (Kronifer) [inline server/connections.py]: Please switch this to return a list of users instead, as it makes sense for users who join later than others as they won't have a way to see all the users

Comment (Kronifer) [reply server/connections.py]: I'm in the process of making this a PR without the container class at the moment.

**Review by Kronifer — CHANGES_REQUESTED** [review]
Some changes are needed, however this is an excellent basis

File: server/connections.py (hunk 2 above)
Comment (ghost) [reply server/connections.py]: Not sure I follow. Do you mean just one method to return a list of all users, or a list ordered by join time? Or to store usernames as they join in a stack, and remove them when they leave?

Comment (Kronifer) [reply server/connections.py]: You could just replace those two methods with a method that returns this: `str(list(CONNECTIONS.data.keys())).replace("'", '"')`

Comment (ghost) [reply server/connections.py]: Oh, I see, it has to be a string representation of current users. I'll change it.

Comment (ghost) [reply server/connections.py]: But how would that allow tracking recently entered or exited usernames?

Comment (Kronifer) [reply server/connections.py]: when we remove a user from the collection, it is removed from the list. and when a user joins or leaves, we would send a new list.

Comment (ghost) [reply server/connections.py]: So it's the whole list that should be sent with the event whether the user leaves or joins, meaning tracking individual users is unimportant?

Comment (Kronifer) [reply server/connections.py]: Yep. This is the easiest way I've found to keep clients updated on all users.

Comment (ghost) [reply server/connections.py]: Ah, sorry I misunderstood. I'll make the changes.

**Review by Kronifer — COMMENTED** [review]
Code looks good, I'll test tomorrow as I have work tonight and if testing is good I'll merge

**Review by Kronifer — APPROVED** [review]
Looks good

**Pull request merged**